x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/SearchableSnapshotAction.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ilm;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.admin.cluster.snapshots.status.SnapshotStatus;
+import org.elasticsearch.action.admin.cluster.snapshots.status.SnapshotsStatusRequest;
+import org.elasticsearch.action.admin.cluster.snapshots.status.SnapshotsStatusResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.cluster.SnapshotsInProgress;
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.TriConsumer;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.snapshots.SnapshotException;
+import org.elasticsearch.xpack.core.ilm.Step.StepKey;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import static org.elasticsearch.cluster.SnapshotsInProgress.State.ABORTED;
+import static org.elasticsearch.cluster.SnapshotsInProgress.State.FAILED;
+import static org.elasticsearch.cluster.SnapshotsInProgress.State.MISSING;
+import static org.elasticsearch.cluster.SnapshotsInProgress.State.SUCCESS;
+
+/**
+ * A {@link LifecycleAction} that will convert the index into a searchable snapshot, by taking a snapshot of the index, creating a
+ * searchable snapshot and the corresponding "searchable snapshot index", deleting the original index and swapping its aliases to the
+ * newly created searchable snapshot backed index.
+ */
+public class SearchableSnapshotAction implements LifecycleAction {
+    public static final String NAME = "searchable_snapshot";
```

Comment:
```suggestion
    public static final String NAME = "searchable-snapshot";
```

Reply:
I think this one may have been lost in some of the other reviews, but I still think it should be changed